lambda/translprim.ml

```diff
@@ -1094,6 +1094,15 @@ let lookup_primitive loc ~poly_mode ~poly_sort pos p =
     | "%unsafe_set_idx" ->
       let layout = List.nth (get_arg_layouts ()) 2 in
       Primitive(Pset_idx (layout, get_first_arg_mode ()), 3)
+    | "%unsafe_get_ptr_imm" ->
+      (* Only safe if the pointer is truly immutable *)
+      Primitive(Pget_ptr (layout, Immutable), 1)
+    | "%unsafe_get_ptr" ->
+      (* [Mutable] is the more conservative version *)
+      Primitive(Pget_ptr (layout, Mutable), 1)
+    | "%unsafe_set_ptr" ->
```

Comment:
I think this file also needs an case for `Pset_ptr` down in `specialize_primitive`, like the one for `Pset_idx`, or we're going to have unnecessary `caml_modify`s.  Probably the test `testsuite/tests/typing-layouts-caml-modify/basics.ml` should also be extended.